sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -115,6 +115,11 @@ final class CastExprMeta[INPUT <: CastBase](
         YearParseUtil.tagParseStringAsDate(conf, this)
       case (_: StringType, _: DateType) =>
         YearParseUtil.tagParseStringAsDate(conf, this)
+      case (_: StringType, dt:DecimalType) =>
+        if (dt.scale < 0 && !ShimLoader.getSparkShims.isNegativeDecimalScaleSupportEnabled) {
+          willNotWorkOnGpu("Rapids doesn't support negative decimal scale for this version of " +
```

Comment:
Now I'm confused.  The test was changed to skip implying it is not a bug, but this text implies it's a problem/limitation in RAPIDS.  Do we expect this to be fixed by RAPIDS in the future?

Reply:
This isn't a problem with RAPIDS but with Spark. The error message is simply saying that we are incompatible with this particular version of Spark. I have added the Spark bug number to the description for more context